hydro_lang/src/stream.rs

```diff
@@ -1573,6 +1574,120 @@ impl<'a, T, L: Location<'a> + NoTick + NoAtomic, B, Order> Stream<T, L, B, Order
         Stream::new(Atomic { tick: tick.clone() }, self.ir_node.into_inner())
     }
 
+    /// Consumes a stream of `Future<T>`, produces a new stream of the resulting `T` outputs.
+    /// Future outputs are produced as available, regardless of input arrival order.
+    ///
+    /// # Example
+    /// ```rust
+    /// # use std::collections::HashSet;
+    /// # use dfir_rs::futures::StreamExt;
+    /// # use hydro_lang::*;
+    /// # tokio_test::block_on(test_util::stream_transform_test(|process| {
+    /// let tick = process.tick();
+    /// process.source_iter(q!([2, 3, 1, 9, 6, 5, 4, 7, 8]))
+    ///     .map(q!(|x| async move {
+    ///         // tokio::time::sleep works, import then just sleep does not, unsure why
+    ///         tokio::time::sleep(tokio::time::Duration::from_millis(10)).await;
+    ///         x
+    ///     }))
+    ///     .resolve_futures()
+    /// #   },
+    /// #   |mut stream| async move {
+    /// // 1, 2, 3, 4, 5, 6, 7, 8, 9
```

Comment:
Can we add a small `(in arbitrary order)` note and scramble the order just so developers don't make any assumptions about that?